formula/src/main/java/com/instacart/formula/internal/FormulaManagerImpl.kt

```diff
@@ -159,29 +173,33 @@ internal class FormulaManagerImpl<Input, State, RenderModel>(
         @Suppress("UNCHECKED_CAST")
         val manager = children
             .findOrInit(key) {
-                childManagerFactory.createChildManager(formula, input, transitionLock)
+                childManagerFactory.createChildManager(formula, input, transitionLock).apply {
+                    setTransitionListener { message, isValid ->
+                        handleTransition(Transition(effects = message), !isValid)
+                    }
+                }
             }
             .requestAccess {
                 throw java.lang.IllegalStateException("There already is a child with same key: $key. Use [key: String] parameter.")
             } as FormulaManager<ChildInput, ChildState, ChildRenderModel>
 
-        manager.setTransitionListener { message, isValid ->
-            handleTransition(Transition(effects = message), !isValid)
-        }
-
         return manager.evaluate(formula, input, processingPass).renderModel
     }
 
     override fun markAsTerminated() {
         terminated = true
         frame?.transitionCallbackWrapper?.terminated = true
-
         callbacks.disableAll()
-        updateManager.terminate()
```

Comment:
`markAsTerminated` is called before we are ready to perform side-effects.  We need to delay `UpdateManager.terminate` until we are ready to handle transitions.